File: src/main/java/org/meetkt/basket/domain/model/Item.java
package org.meetkt.basket.domain.model;

import org.meetkt.catalogue.domain.model.Product;
import org.meetkt.catalogue.domain.model.ProductId;
import org.meetkt.display.domain.model.Price;

import java.util.Objects;

public class Item {

    private final ProductId productId;
    private final Price price;

    public Item(Product product) {
        Objects.requireNonNull(product);

        this.productId = product.id();
        this.price = product.price();
    }

    //TODO: Think of a better name
    public boolean contains(ProductId productId) {
        return this.productId.equals(productId);
    }

    public Price price() {
        return price;
    }

    public ProductId productId() {
        return productId;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Item)) return false;
        Item item = (Item) o;
        return Objects.equals(productId, item.productId);
    }

    @Override
    public int hashCode() {
        return productId.hashCode();
    }
}

File: src/main/kotlin/in/meetkt/basket/domain/model/Item.kt
package `in`.meetkt.basket.domain.model

import `in`.meetkt.catalogue.domain.model.Product
import `in`.meetkt.catalogue.domain.model.ProductId

class Item(product: Product) {

    val productId = product.id
    val price = product.price

    fun contains(productId: ProductId): Boolean {
        return this.productId == productId
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Item
        if (productId != other.productId) return false
        return true
    }

    override fun hashCode(): Int {
        return productId.hashCode()
    }

    override fun toString(): String {
        return "Item(productId=$productId, price=$price)"
    }
}